app/controllers/miq_ae_class_controller.rb

```diff
@@ -389,50 +389,36 @@ def replace_right_cell(replace_trees = [])
     render :js => presenter.to_html
   end
 
-  def get_combo_xml(fields)
+  def build_type_options(fields)
     aetypes = MiqAeField.available_aetypes
-    combo_xml = Array.new
+    combo_options = []
     fields.each do |fld|
-      xml = REXML::Document.load("")
-      xml << REXML::XMLDecl.new(1.0, "UTF-8")
-      # Create root element
-      root = xml.add_element("complete")
-      if fld['aetype'].blank? && (session[:field_data].blank? || session[:field_data][:aetype].blank?)
+      if fld['aetype'].blank? && session.fetch_path(:field_data, :aetype).blank?
         fld['aetype'] = "attribute"
-      elsif !session[:field_data].blank? && !session[:field_data][:aetype].blank?
-        fld['aetype'] = session[:field_data][:aetype]
+      elsif !session.fetch_path(:field_data, :aetype).blank?
+        fld['aetype'] = session.fetch_path(:field_data, :aetype)
       end
-      aetypes.each do |aetype|
-        opt = root.add_element("option", {"value"=>aetype,"img_src"=>"/images/icons/new/16_ae_#{aetype}.png"})
-        opt.add_attribute("selected", "true") if fld['aetype'] == aetype
-        opt.text = aetype.titleize
-      end
-      combo_xml.push(xml.to_s)
     end
-    return combo_xml
+    aetypes.each do |aetype|
+      combo_options.push([aetype.titleize, aetype, {"data-icon" => "product product-#{aetype}"}])
+    end
+    combo_options
```

Comment:
This could also be simplified to:

```ruby
  aetypes.collect { |aetype| [aetype.titleize, aetype, {"data-icon" => "product product-#{aetype}"}] }
```

replacing

```ruby
  combo_options = []
  aetypes.each do |aetype|
    combo_options.push([aetype.titleize, aetype, {"data-icon" => "product product-#{aetype}"}])
  end
  combo_options
```